mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMailboxManager.java

```diff
@@ -507,6 +512,7 @@ private void validateDestinationPath(MailboxPath newMailboxPath, MailboxSession
         if (mailboxExists(newMailboxPath, session)) {
             throw new MailboxExistsException(newMailboxPath.toString());
         }
+        assertIsOwner(session, newMailboxPath);
```

Comment:
is it really needed? What if the user has write access on this mailbox without being the owner?

Reply:
Hmmm I tend to disagree with that. 

I think the mailboxPath here should belong to the user because `validateDestinationPath` is only used by the `renameMailbox` methods. I believe user1 should not be able to rename his mailbox to a mailbox for user2 (thus resulting in an ownership transfert?), with the added tests `renameMailboxShouldThrowWhenToMailboxPathDoesNotBelongToUser` and `renameMailboxByIdShouldThrowWhenToMailboxPathDoesNotBelongToUser` being here to enforce this behavior.

The destination mailbox, as it's a rename, is supposed to not exist yet, so impossible for the user to have write access on it as well. We could argue that the name `assertIsOwner` is not really correct then, but that part is being changed in #3038 in a more generic way with system and user sessions (see `assertSessionHasAdministrativeAccessOn`).

Does it answer your question? :)

Reply:
As I recall it, this was a temporary decision to mitigate issues not covered by the current ACL system. These actions can not be triggered via IMAP nor via JMAP.

Reply:
thank you for your explanation, it seems fair at first sight. But imagine the following scenario:
- user a delegates all her mailbox rights to user b
- user b wants to rename a mailbox from user a on behalf of user a

In this case the new mailbox name should still belong to user a, no?

Reply:
> In this case the new mailbox name should still belong to user a, no?

We disabled such actions in JMAP. So there's no need of piggy-backing such a complex feature on this code change.

Reply:
ok, do we have integration tests showing that?

Reply:
I didn't investigate much on the JMAP level, but it's an interesting point of view. But then your use-case was not possible in the first place then, with the actual state of the code, as : 
* yes there was no check on user mailbox path ownership for the destination mailbox
* but there is one on the mailbox that user wants to rename... thus user b wouldn't be able to rename a mailbox of user a in the first place

It raises questions though on what should be possible or not for a user session in such corner cases. Could it be perhaps raised in a separate ticket and having some brainstorming on it for a polish sprint or something similar? Would that be acceptable?

Reply:
See `MailboxCreationAndSharing.feature`, `MailboxDeletionAndSharing.feature`, `MoveMailboxAndSharing.feature`

Conclusion in JMAP a sharee can not create, delete nor Move a mailbox. We may lack tests for simple renames though...

Reply:
fine, thank you